pkgs/top-level/module/module.nix

```diff
@@ -0,0 +1,174 @@
+/*
+  A [module] for invoking Nixpkgs.
+
+  This module is a public interface for use in third party module system
+  applications, so that they can re-expose the Nixpkgs configuration logic to
+  their users.
+
+  It declares its options without a prefix like `nixpkgs.*`, so it is meant to
+  be imported in a submodule.
+
+  This module is based on the NixOS `misc/nixpkgs.nix` module, but it gets rid
+  of legacy options. It may be used to replace that module after legacy options
+  have been deprecated and removed.
+
+  As this module declares an option named `config`, you're recommended to pass
+  `shorthandOnlyDefinesConfig = true` to `types.submoduleWith`.
+
+  module: https://nixos.org/manual/nixpkgs/unstable/#module-system
+*/
+{ lib, config, options, ... }:
+
+let
+  inherit (lib) mkOption types literalExpression;
+
+  optDefaultPrio = (lib.mkOptionDefault null).priority;
+
+  isCross = config.buildPlatform != config.hostPlatform;
+  systemArgs =
```

Comment:
Let's have a comment here mentioning that this is a bit of a hack and should be moved into the pkgs entrypoint function definition